src/python/pants/backend/python/util_rules/pex.py

```diff
@@ -181,9 +187,50 @@ def cmp_constraints(req1: Requirement, req2: Requirement) -> int:
             key=functools.cmp_to_key(cmp_constraints),
         )
 
+    @classmethod
+    def resolve_conflicting_fields(
+        cls,
+        deprecated: PythonInterpreterCompatibility,
+        new: InterpreterConstraintsField,
+        address: Address,
+    ) -> Union[PythonInterpreterCompatibility, InterpreterConstraintsField]:
+        if deprecated.value and new.value:
+            raise InvalidFieldException(
+                f"Specified both the deprecated `{deprecated.alias}` field and the new "
+                f"`{new.alias}` field for the target {address}. Please use only one "
+                f"(preferably {new.alias})"
+            )
+        if deprecated.value:
+            return deprecated
+        return new
+
+    @classmethod
+    def create_from_targets(
+        cls, targets: Iterable[Target], python_setup: PythonSetup
+    ) -> "PexInterpreterConstraints":
```

Comment:
This is useful regardless of this PR. We were duplicating a pattern in several places.